services/venice-controller/src/main/java/com/linkedin/venice/controller/DeferredVersionSwapService.java

```diff
@@ -169,41 +202,26 @@ private void logMessageIfNotRedundant(String message) {
    */
   private boolean isDavinciStore(String cluster, Set<String> regions, String storeName, int targetVersionNum) {
     for (String region: regions) {
-      StoreResponse targetRegionStoreResponse = getStoreForRegion(cluster, region, storeName);
+      Version version = getVersionFromStoreInRegion(cluster, region, storeName, targetVersionNum);
```

Comment:
This method should be called in during push start and trim the target region so they falback into  old target region push flow